### PR TITLE
comment out gene list log, add more logging to debug

### PR DESCRIPTION
**Pull request by annacuomo**
Remove logging statement printing out all genes, added logging statements to try debugging why the driver took so long (4 hours: https://batch.hail.populationgenomics.org.au/batches/407644/jobs/1) vs approx 40 minutes before the latest changes (https://batch.hail.populationgenomics.org.au/batches/396869/jobs/1)

Slack convo for context: https://centrepopgen.slack.com/archives/C018KFBCR1C/p1671046610691089?thread_ts=1669709793.232589&cid=C018KFBCR1C

### CODE REVIEW

**Comment by lgruen** [comment]
@MattWellie uses this logging format (includes timestamps):

```python
    logging.basicConfig(
        level=logging.INFO,
        format='%(asctime)s %(levelname)s %(module)s:%(lineno)d - %(message)s',
        datefmt='%Y-%m-%d %H:%M:%S',
        stream=sys.stderr,
    )
```

File: batch.py [hunk 1]
```diff
@@ -706,6 +706,7 @@ def crm_pipeline(
         )
         _plink_genes |= set(extract_genes(genes_of_interest, expression_tsv_path))
     plink_genes = list(sorted(_plink_genes))
+    logging.info(f'Done selecting genes, total number: {len(plink_genes)}')
```
Comment (lgruen) [inline batch.py:719]: Can you add a log statement right before _and_ after each `extract_genes` and `glob`?

**Review by lgruen — CHANGES_REQUESTED** [review]

File: batch.py (hunk 1 above)
Comment (annacuomo) [reply batch.py:719]: absolutely!

Comment (annacuomo) [reply batch.py:719]: let me know if i missed any!

File: batch.py [hunk 2]
```diff
@@ -759,17 +772,20 @@ def manage_concurrency_for_job(job: hb.job.Job):
...
+        logging.info(f'after extracting {celltype}-expressed genes - run association')
+        # logging.info(f'Genes to run: {genes_list}')
         if len(genes_list) == 0:
             logging.info('No genes to run, exit!')
             continue
 
         gene_run_jobs = []
 
         cell_type_root = output_path(celltype)
+        logging.info('before glob (pv files)')
```
Comment (lgruen) [inline batch.py]: Do you want to add the `celltype` here too?

File: batch.py [hunk 3]
```diff
@@ -759,17 +772,20 @@ def manage_concurrency_for_job(job: hb.job.Job):
...
         existing_files = list(to_path(cell_type_root).glob('*_pvals.csv'))
+        logging.info('after glob (pv files)')
```
Comment (lgruen) [inline batch.py]: Maybe print `len(existing_files)` here as well.

File: batch.py [hunk 4]
```diff
@@ -722,7 +733,9 @@ def manage_concurrency_for_job(job: hb.job.Job):
     # submit a job for each gene (export genotypes to plink)
     dependencies_dict: Dict[str, hb.job.Job] = {}
     plink_root = output_path('plink_files')
+    logging.info('before glob (bim files)')
     bim_files = list(to_path(plink_root).glob('*.bim'))
+    logging.info('after glob (bim files)')
```
Comment (lgruen) [inline batch.py]: Maybe print `len(bim_files)` here as well.

File: batch.py [hunk 5]
```diff
@@ -759,17 +772,20 @@ def manage_concurrency_for_job(job: hb.job.Job):
                 f'{celltype}_expression.tsv',
             )
         )
-
+        logging.info(f'before extracting {celltype}-expressed genes - run association')
         genes_list = extract_genes(genes_of_interest, expression_tsv_path)
-        logging.info(f'Genes to run: {genes_list}')
+        logging.info(f'after extracting {celltype}-expressed genes - run association')
```
Comment (lgruen) [inline batch.py]: Maybe print `len(genes_list)` here as well.

File: batch.py [hunk 6]
```diff
@@ -481,7 +487,9 @@ def summarise_association_results(
     """
     from multipy.fdr import qvalue
 
+    logging.info('before glob (pv files) - summarise job')
     existing_pv_files = list(to_path(pv_folder).glob('*_pvals.csv'))
+    logging.info('after glob (pv files) - summarise job')
```
Comment (lgruen) [inline batch.py]: Maybe print `len(existing_pv_files)` here as well.

**Review by lgruen — APPROVED** [review]

File: batch.py [hunk 7]
```diff
@@ -759,17 +772,24 @@ def manage_concurrency_for_job(job: hb.job.Job):
...
+        logging.info(
+            f'before extracting {celltype}-expressed genes to run association for'
+        )
         genes_list = extract_genes(genes_of_interest, expression_tsv_path)
-        logging.info(f'Genes to run: {genes_list}')
+        logging.info(
+            f'after extracting {celltype}-expressed genes: run association for {len(genes_list)} genes'
+        )
+        # logging.info(f'Genes to run: {genes_list}')
         if len(genes_list) == 0:
```
Comment (lgruen) [inline batch.py]:
```suggestion
        if not genes_list:
```

**Review by lgruen — APPROVED** [review]

**Pull request merged**